### PR TITLE
Fix missing SDK version for test building it GitHub Workflow

### PR DESCRIPTION
**Pull request by Turnerj**
See the build/test error in #1328 - this *might* patch around that neatly via using the matrix options. This PR is as much of a test of the concept.

### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Uses the options matrix to define the SDK version to install, thus hopefully guaranteeing the right version is installed for tests.

### CODE REVIEW

**Comment by Turnerj** [comment]
Slight issue with it - seems that the `Benchmarks` project and the `ProfilingSandbox` project are being built but require specific versions of the SDK themselves. The builds might need to be more selective to avoid building either project as neither is used for CI AFAIK.

**Comment by JimBobSquarePants** [comment]
I think we have to install both for each build now.

https://github.com/actions/setup-dotnet

**Comment by Turnerj** [comment]
Hmmmm, I was thinking going the other way, changing the `ci-build.ps1` script to just build the main library and the test library.
https://github.com/SixLabors/ImageSharp/blob/d95a996ed5ba14a1421dafeb844a56ab08211ead/ci-build.ps1#L11

If that was building the test library, the test library would automatically build the main library.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -55,7 +60,7 @@ jobs:
             - name: Setup DotNet SDK
               uses: actions/setup-dotnet@v1
               with:
-                  dotnet-version: "3.1.x"
+                  dotnet-version: ${{matrix.options.sdkVersion}}
```
Comment (JimBobSquarePants) [inline .github/workflows/build-and-test.yml:63]: I can't figure out from the docs whether you can simply do. [ "2.1.x", "3.1.x" ]

Comment (Turnerj) [reply .github/workflows/build-and-test.yml:63]: Digging through the TypeScript code for the action, it looks to only take string values. Internally, the value is getting passed into here: https://github.com/actions/setup-dotnet/blob/e3d81d94533f1e31c11d94a1fdf6f60f096a7bc5/src/installer.ts#L33

Comment (JimBobSquarePants) [reply .github/workflows/build-and-test.yml:63]: Yeah, ran a quick test  in the Web repo. Big fail

**Comment by Turnerj** [comment]
I need to go get some sleep - I've pushed a change which tweaks what is built for the CI which may work around the issue. Its cool if this isn't the direction you want to go though.

Hopefully it works!

**Comment by tocsoft** [comment]
I don't think we need the "Setup DotNet SDK" step at all anymore, I believe all the sdks we need are all preinstalled on the images... we needed it in the past but I don't think we do anymore.

**Comment by JimBobSquarePants** [comment]
@tocsoft You're right!

**Comment by JimBobSquarePants** [comment]
> I don't think we need the "Setup DotNet SDK" step at all anymore, I believe all the sdks we need are all preinstalled on the images... we needed it in the past but I don't think we do anymore.

@tocsoft You're right!

**Comment by JimBobSquarePants** [comment]
Thanks for your help @Turnerj 

I ended up just pushing the fix in the other open PR. 

Turns out the new VMs have everything!

https://github.com/actions/virtual-environments/releases

**Comment by Turnerj** [comment]
Awesome - glad you got it working!

**Pull request closed without merging**